[FEATURE] Add Arch-specific defconfig

Tashar02 opened the pull request:
This defconfig has been taken from the official archlinux kernel and has been adapted to the changes you (dakkshesh07) have made to your fedora_defconfig and has been regenerated afterwards. The changes are segmented into 3 commits for better understanding and history.
Please test it, review it and check if something has gone missing which wasn't supposed to be.

Regards,
Tashar.

Tashar02 committed:
x86/configs: Import Arch defconfig

This is a clone of the config [1] used for archlinux.

[1]: https://github.com/archlinux/svntogit-packages/blob/packages/linux/trunk/config

Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>

Tashar02 committed:
x86/configs: arch: Regenerate

Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>

Tashar02 committed:
x86/configs: arch: Sync modifications by @dakkshesh07

This syncs and adapt changes by @dakkshesh07 till this date [1] from
fedora_defconfig to arch_defconfig excluding some changes that are
unnecessary and including a regeneration.

[1]: https://github.com/Neutron-Projects/kernel-dejavu/blob/6a233dbc32079d3b89b115234b56f499497d7180/arch/x86/configs/fedora_defconfig

Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>

beakthoven reviewed (commented):
All of the changes appear to be good. Could you confirm that you built with the defconfig and tried booting it?

Tashar02 committed:
kbuild: Import & tweak PKGBUILD for our needs

This PKGBUILD has been taken from cybernight777's repo [1] and tweaked
by me and dakkshesh07 for our needs.

[1] https://github.com/cyberknight777/linux-dragonheart/commits/v6.0/PKGBUILD

Co-authored-by: Cyber Knight <cyberknight755@gmail.com>
Co-authored-by: Dakkshesh <dakkshesh5@gmail.com>
Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>

Tashar02 committed:
cirrus: Instantiate CirrusCI

Inspired from: https://github.com/cyberknight777/linux-dragonheart/blob/v6.0/.cirrus.yml

- Use tab indentations.
- Use wget instead of curl.
- Use bash cmd directly instead of "./"
- Use katbin instead of Cyber's rustbin.
- Convert bash script to unix before executing

Co-authored-by: Cyber Knight <cyberknight755@gmail.com>
Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>

Tashar02 committed:
cirrus: Clone neutron clang from yml

Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>
